### ISSUE TITLE
Add support for Crystal

### ISSUE THREAD

**Issue by watzon** [dependabot/dependabot-core#1237]
Crystal, for those who don't know, is a statically typed and compiled language with a syntax very similar to Ruby. For dependencies it uses a file called `shard.yml`, the spec for which can be found [here](https://github.com/crystal-lang/shards/blob/master/SPEC.md). Hopefully adding support won't be overly difficult.

**Comment by greysteil** [comment]
Totally 👍 on adding Crystal support.

**Comment by alexanderadam** [comment]
No need to close it. It's still relevant. 😉

**Comment by watzon** [comment]
Agreed. We need to get some more community support behind this though

**Comment by rebelagentm** [comment]
👋 The Dependabot team is swamped scaling for GitHub at the moment, so it will, unfortunately, be a while before we can get to adding enhancements like this.  If someone wants to submit a PR adding support for this, we'll be happy to take review it as soon as we can.

**Comment by watzon** [comment]
I figured that would be the case. Hopefully someone looking for a project to help the Crystal community out will be willing to jump in and give it a go

**Comment by alexanderadam** [comment]
I've setup [a test repository with outdated dependencies](https://github.com/alexanderadam/renovatebot_crystal_test) in case you need something to test.
I created it for [the corresponding renovatebot issue](https://github.com/renovatebot/renovate/issues/6029#issuecomment-618251989) but dependabot folks naturally can work with it as well.

**Comment by caspiano** [comment]
Any progress on this? I'd be happy to attempt to contribute!

**Comment by jurre** [comment]
Appreciate the offer @caspiano but we've currently [put a pause on adding new ecosystems to dependabot-core](https://github.com/dependabot/dependabot-core/blob/05a2faf57fbaf3f3a6954a61cdf1c76d88a0c3c3/CONTRIBUTING.md#contributing-new-ecosystems) as we beef up support for the current offering.

**Comment by caspiano** [comment]
Bummer! Nonetheless, I can't wait 😊

**Comment by grepsedawk** [comment]
It's been awhile since the beef up, if there's anything I can do to help out please do reach out :)

**Comment by alexanderadam** [comment]
> Hi there, still no news about Crystal support? :)

I have the feeling that it's more likely to have [Renovate being able to handle Crystal](https://github.com/renovatebot/renovate/issues/6029).

But let's see :crossed_fingers:

**Comment by Blacksmoke16** [comment]
In light of https://github.com/dependabot/dependabot-core/pull/8844, does that imply a PR would be accepted to support `shards`?

**Comment by jwoertink** [comment]
Would be neat to see this added

**Comment by Blacksmoke16** [comment]
I been working on this on-and-off for a little while: https://github.com/dependabot/dependabot-core/compare/main...Blacksmoke16:dependabot-core:shards-support. Def would be open to handing it off to someone else, or pairing up to get it over the finish line.
